### PR TITLE
Variable names, Test Updates, Reformats Oh my

### PR DESCRIPTION
**Pull request by jthurman42**
Changes to most commands for variable names, improved testing, and syntax changes for the commands.

### CODE REVIEW

File: internal/apm/command_application.go [hunk 1]
```diff
@@ -7,85 +7,109 @@ import (
...
-	Use:   "get",
-	Short: "Get a New Relic application",
-	Long: `Get a New Relic application
+var cmdAppSearch = &cobra.Command{
+	Use:   "search",
+	Short: "Search for a New Relic application",
+	Long: `Search for a New Relic application
 
-The get command performs a query for an APM application by ID, entity GUID, name, and/or account ID.
+The search command performs a query for an APM application name, and/or account ID.
```
Comment (ctrombley) [inline internal/apm/command_application.go]: nit: remove the comma after `name`?

File: internal/apm/command_application.go [hunk 2]
```diff
@@ -7,85 +7,109 @@ import (
...
 `,
-	Example: "newrelic apm application get --name <appName>",
+	Example: "newrelic apm application search --name <appName>",
 	Run: func(cmd *cobra.Command, args []string) {
 
-		if apmApplicationName == "" && apmApplicationAccountID == "" && apmApplicationGUID == "" {
-			log.Fatal("one of --name, --acountId or --guid are required")
+		if appName == "" && appAccountID == "" {
+			log.Fatal("one of --name or --acountId are required")
```
Comment (ctrombley) [inline internal/apm/command_application.go]: typo: `acountId`

File: internal/apm/command_application.go [hunk 3]
```diff
@@ -7,85 +7,109 @@ import (
...
 func init() {
-	Command.AddCommand(apmApplication)
-	apmApplication.AddCommand(apmGetApplication)
-	apmGetApplication.Flags().IntVarP(&apmApplicationID, "applicationId", "a", 0, "search for results matching the given APM application ID")
-	apmGetApplication.Flags().StringVarP(&apmApplicationName, "name", "n", "", "search for results matching the given APM application name")
-	apmGetApplication.Flags().StringVarP(&apmApplicationGUID, "guid", "g", "", "search for results matching the given APM application GUID")
-	apmGetApplication.Flags().StringVarP(&apmApplicationAccountID, "accountId", "", "", "search for results matching the given APM application account ID")
+	Command.AddCommand(cmdApp)
+
+	cmdApp.PersistentFlags().IntVarP(&appID, "applicationId", "a", 0, "search for results matching the given APM application ID")
```
Comment (ctrombley) [inline internal/apm/command_application.go]: 👍

File: internal/apm/command_deployment.go [hunk 4]
```diff
@@ -4,134 +4,118 @@ import (
...
+
+	cmdDeployment.AddCommand(cmdDeploymentCreate)
+	cmdDeploymentCreate.Flags().StringVarP(&deployment.Description, "description", "", "", "the description stored with the deployment")
+	cmdDeploymentCreate.Flags().StringVarP(&deployment.User, "user", "", "", "the user creating with the deployment")
+	cmdDeploymentCreate.Flags().StringVarP(&deployment.Changelog, "change-log", "", "", "the change log stored with the deployment")
+
+	cmdDeploymentCreate.Flags().IntVarP(&appID, "applicationId", "a", 0, "the application ID the deployment will be created for")
+	utils.LogIfError(cmdDeploymentCreate.MarkFlagRequired("applicationId"))
+
+	cmdDeploymentCreate.Flags().StringVarP(&deployment.Revision, "revision", "r", "", "a freeform string representing the revision of the deployment")
```
Comment (ctrombley) [inline internal/apm/command_deployment.go]: 👍

File: internal/entities/command_search.go [hunk 5]
```diff
@@ -72,9 +72,17 @@ The search command performs a search for New Relic entities.
 			if len(entityFields) > 0 {
 				mapped := mapEntities(entities, entityFields, utils.StructToMap)
 
-				json, err = prettyjson.Marshal(mapped)
+				if len(mapped) == 1 {
```
Comment (ctrombley) [inline internal/entities/command_search.go]: This conditionally changes the return type.  Could that cause problems for users?

Comment (jthurman42) [reply internal/entities/command_search.go]: The argument here was that we should not return an array for a single item result.  We've discussed both sides of it, I don't have a strong leaning either way, but put it in anyway.

**Pull request merged**